src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -117,70 +117,27 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
-    //   dPorosity_dPressure += fixedStressPressureCoefficient;
-    //   dPorosity_dTemperature += fixedStressTemperatureCoefficient;
-    // }
-
-    // using total stress formulation and match the other formulation
-    // real64 ttlStrs = meanEffectiveStressIncrement_k -  biotCoefficient * (deltaPressureFromBeginningOfTimeStep - deltaPressureFromLastIteration);
-    // porosity = porosity_n
-    //           + biotCoefficient * ttlStrs / bulkModulus // change due to stress increment (at the previous
-    //                                                                             // sequential iteration)
-    //           + biotSkeletonModulusInverse * deltaPressureFromBeginningOfTimeStep // change due to pressure increment
-    //           - porosityThermalExpansion * deltaTemperatureFromBeginningOfTimeStep; // change due to temperature increment
-    // dPorosity_dPressure = biotSkeletonModulusInverse;
-    // dPorosity_dTemperature = -porosityThermalExpansion;
-  
-    // if( !isZero( meanEffectiveStressIncrement_k ) ) // TODO: find a better way to disable this at the first flow iteration
-    // {
-    //   porosity += fixedStressPressureCoefficient * deltaPressureFromBeginningOfTimeStep // fixed-stress pressure term
-    //               + fixedStressTemperatureCoefficient * deltaTemperatureFromLastIteration; // fixed-stress temperature term
-    //   dPorosity_dPressure += fixedStressPressureCoefficient;
-    //   dPorosity_dTemperature += fixedStressTemperatureCoefficient;
-    // }
-
-    //accelerated total strs
-    static int output = 0;
-    real64 ttlStrs = meanEffectiveStressIncrement_k;
-    if (!isZero(ttlStrs) && !output ){
-      std::cout << bulkModulus <<" " << meanEffectiveStressIncrement_k << " " << biotCoefficient << " " << deltaPressureFromBeginningOfTimeStep << std::endl;
-      output = 1;
-    }
+    // total stress formulation for porosity update
     porosity = porosity_n
-              + biotCoefficient * ttlStrs / bulkModulus // change due to stress increment (at the previous
+              + biotCoefficient * averageMeanTotalStressIncrement_k / bulkModulus // change due to stress increment (at the previous
                                                                                 // sequential iteration)
               + biotSkeletonModulusInverse * deltaPressureFromBeginningOfTimeStep // change due to pressure increment
               - porosityThermalExpansion * deltaTemperatureFromBeginningOfTimeStep; // change due to temperature increment
     dPorosity_dPressure = biotSkeletonModulusInverse;
     dPorosity_dTemperature = -porosityThermalExpansion;
-
-    if( !isZero( meanEffectiveStressIncrement_k ) ) // TODO: find a better way to disable this at the first flow iteration
+  
+    if( !isZero( averageMeanTotalStressIncrement_k ) ) // TODO: find a better way to disable this at the first flow iteration
```

Comment:
i think in the first iteration of the first time step, this number is non-zero (since pressure is already calculated). Is the fixedStressPressureCoefficient already defined?

Reply:
Was this ever resolved? @paveltomin

Reply:
I mean, is averageMeanTotalStressIncrement_k zero on the first iteration of a given time step? Guotong is right, the fixedStressPressureCoefficient can be wrong on the first iteration.

Reply:
> Guotong is right, the fixedStressPressureCoefficient can be wrong on the first iteration.

I am thinking to address that in a small separate PR, do you have a draft for fix?

Reply:
created this https://github.com/GEOS-DEV/GEOS/pull/2766 to address the specific issue with that magic if condition